burnman/planet.py

```diff
@@ -7,206 +7,834 @@
 
-        # we need to skip scaling gravity[0]
-        self.gravity[1:] = self.gravity[1:]/self.radial_slices[1:]/self.radial_slices[1:]
+        return np.array(grav)
 
-    def _compute_pressure(self, density, gravity, radii):
+    def _compute_pressure(self, density, gravity, pressure_top):
         """
         Calculate the pressure profile based on density and gravity.  This integrates
         the equation for hydrostatic equilibrium  P = rho g z.
+        Used to update the pressure profile in _evaluate_eos()
         """
-
-        start_pressure = 0.0
+        start_pressure = pressure_top
+        press = []
         for layer in self.layers[::-1]:
-            radii = self.radial_slices[layer.n_start: layer.n_end]
-            density = self.densities[layer.n_start: layer.n_end]
-            gravity = self.gravity[layer.n_start: layer.n_end]
-
-            # convert radii to depths
-            depths = radii[-1]-radii
-
-            # Make a spline fit of density as a function of depth
-            rhofunc = UnivariateSpline(depths[::-1], density[::-1])
-            # Make a spline fit of gravity as a function of depth
-            gfunc = UnivariateSpline(depths[::-1], gravity[::-1])
-
-            # integrate the hydrostatic equation
-            pressure = np.ravel(odeint((lambda p, x : gfunc(x)* rhofunc(x)), start_pressure, depths[::-1]))
-            start_pressure = pressure[-1]
-
-            self.pressures[layer.n_start: layer.n_end] = pressure[::-1]
+            press.extend(layer._compute_pressure(
+                                                 density[layer.n_start: layer.n_end], gravity[layer.n_start: layer.n_end], start_pressure)[::-1])
+            start_pressure = press[-1]
+        return np.array(press)[::-1]
 
-    def _compute_mass( self):
+    @property
+    def mass(self):
         """
         calculates the mass of the entire planet [kg]
         """
         mass = 0.0
         for layer in self.layers:
-            radii = self.radial_slices[layer.n_start: layer.n_end]
-            density = self.densities[layer.n_start: layer.n_end]
-            rhofunc = UnivariateSpline(radii, density)
-            layer.mass = quad(lambda r : 4*np.pi*rhofunc(r)*r*r,
-                            radii[0], radii[-1])[0]
             mass += layer.mass
         return mass
 
-    def _compute_moment_of_inertia( self):
+    @property
+    def moment_of_inertia(self):
         """
         #Returns the moment of inertia of the planet [kg m^2]
         """
         moment = 0.0
         for layer in self.layers:
-            radii = self.radial_slices[layer.n_start: layer.n_end]
-            density = self.densities[layer.n_start: layer.n_end]
-            rhofunc = UnivariateSpline(radii, density)
-            moment += quad(lambda r : 8.0/3.0*np.pi*rhofunc(r)*r*r*r*r,
-                           radii[0], radii[-1])[0]
+            moment += layer.moment_of_inertia
         return moment
```

Comment:
This function and the ones for gravity and mass above are a bit cumbersome. How about:
`return np.sum([layer.moment_of_inertia for layer in self.layers])`

(There's actually a lot more in this PR that could be made more readable with list comprehensions, but the ones starting with empty lists or sums starting from 0.0 caught my eye.)